### ISSUE TITLE
skip parameter not taking expression for boolean

### ISSUE THREAD

**Issue by wangsongiam** [dotansimha/graphql-code-generator#2145]
**Describe the bug**
 

When using version `1.3.1` with `withHooks`, the input type for render props query component is not taking `boolean` type for `skip` param type, instead, I see `{ variables: {...}; skip?: false } | { skip: true }`. So skip can only be set with `true` or `false`. 

This will break common expression like `!undefined` or `!a && b`. 

eg.
```
<Get_FooComponent variables={{...}} skip={!a && !b}>
```

**To Reproduce**

1. My GraphQL schema:

```graphql
# Put your schema here
```

2. My GraphQL operations:

```graphql
# Put your operations here
```

3. My `codegen.yml` config file:

```yml
# Put your YML here
generates:
  src/_generated/graphql.tsx:
    documents:
      - src/components/**/*.{ts,tsx}
      - src/queries/**/*.{ts,tsx}
    plugins:
      - 'typescript'
      - 'typescript-operations'
      - 'typescript-react-apollo'
    config:
      withHooks: true

```

**Expected behavior**

**Environment:**

- OS: 
- `@graphql-codegen/...`: 
- NodeJS: 

**Additional context**

**Comment by ardatan** [comment]
I didn't understand the problem, so `skip` should take a boolean which can be the result of an expression.

**Comment by wangsongiam** [comment]
it seems that `skip` can only be set to true or false, not the result of an expression. So, `<Get_FooComponent variables={{...}} skip={true}> ` works but `<Get_FooComponent variables={{...}} skip={!false}> ` wouldn't

**Comment by ardatan** [comment]
What do you mean by not working? Is there any TS error or isn't it working on runtime?

**Comment by wangsongiam** [comment]
there's a TS error says
```
Types of property 'skip' are incompatible.
Type 'boolean' is not assignable to type 'true'.ts(2322)
```

**Comment by palindrom615** [comment]
that's right. these are part of generated code:

```typescript
export type PartnerPayoutAccountsComponentProps = Omit<
  ReactApollo.QueryProps<
    PartnerPayoutAccountsQuery,
    PartnerPayoutAccountsQueryVariables
  >,
  "query"
> & // totally not needed part: why was the code generated?
  (
    | { variables: PartnerPayoutAccountsQueryVariables; skip?: false }
    | { skip: true });
```

and above code prevent from assign boolean value into `skip`.

**Comment by wangsongiam** [comment]
If this is intended, feel free to close. But I wonder if there's any reason for avoiding boolean type here? Not sure if I'm the only one but I was using an expression for `skip`.

**Comment by palindrom615** [comment]
https://github.com/dotansimha/graphql-code-generator/pull/1926/files?file-filters%5B%5D=.ts#diff-ce84bc42a6b0c5a7f856c37781eb4e65R113

This commit cause the issue. But, is it really good not to let assign boolean value into skip parameter?

```
isSkip: boolean = true;
variables = {};
<GeneratedComponent
  variables={isSkip ? undefined : variables}
  skip={isSkip}
/>
```

Above code looks fair to me but cannot be compiled because typescript cannot check type of the code from generated type definitions as 

```
(
    | { variables: GeneratedVariables; skip?: false }
    | { skip: true });
```

**Comment by palindrom615** [comment]
I reverted the commit and made pr. Letting variable optional when skip parameter is true is quite awesome, but it is meaningless when type check cannot catch.

**Comment by dotansimha** [comment]
Fixed in: https://github.com/dotansimha/graphql-code-generator/pull/2251

**Comment by wangsongiam** [comment]
awesome! Thanks, guys!

**Comment by dotansimha** [comment]
@wangsongiam keeping open until we'll release a stable version with the fix :)

**Comment by dotansimha** [comment]
Fixed in 1.5.0 🎉
